mos_tests/object_storage/conftest.py

```diff
@@ -0,0 +1,255 @@
+def swift_create_container(swift_client):
+    """Creates container with Swift client"""
+    bucketname = 'TESTBUCKET{0}'.format(random.randint(0, 10000))
+    out = swift_client.post(bucketname)
+    yield bucketname, out
+    # remove buckets if it wasn't done in test
+    try:
+        swift_client.delete(bucketname)
+    except Exception:
+        pass
+
+
+@pytest.yield_fixture
+def s3cmd_create_container(s3cmd_client):
+    """Creates container with S3cmd client"""
+    bucketname = 'TESTBUCKET{0}'.format(random.randint(0, 10000))
+    out = s3cmd_client.bucket_make(bucketname)
+    yield bucketname, out
+    # remove buckets if it wasn't done in test
+    try:
+        s3cmd_client.bucket_remove(bucketname, recursive=True)
+    except Exception:
+        pass
+
+
+@pytest.yield_fixture
+def create_file_on_node(ctrl_remote, request):
+    """Creates tmp file with requested size"""
+    try:
+        size_mb = request.param
+    except Exception:
+        size_mb = 111  # MB
+    _, f_path = tempfile.mkstemp(prefix='ObjStor_')
+    f_name = ntpath.basename(f_path)
+    cmd = 'fallocate -l {0}M {1}'.format(size_mb, f_path)
+    ctrl_remote.check_call(cmd)
+    yield f_path, f_name
+    # delete file
+    cmd = r'\rm -rf {0}'.format(f_path)
+    ctrl_remote.check_call(cmd)
+
+
+@pytest.yield_fixture
+def s3cmd_install_configure(env, ctrl_remote, openstack_client, admin_remote):
+    """Install and configure s3cmd on controller"""
+
+    ceph_config_file = '/etc/ceph/ceph.conf'         # cfg file on controller
+    restart_radosgw = "/etc/init.d/radosgw restart"  # restart command
+    remote_cfg_file = '/root/.s3cfg'                 # s3cmd cfg file on node
+    local_templ_name = 's3cfg'                       # local template name
+
+    def templates_dir_path():
+        """Returns full path to local template dir"""
+        return os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), 'templates/')
+
+    def get_management_vip_ip(admin_remote):
+        """Get 'management_vip' from fuel network config"""
+        cmd = "fuel --env 1 network --download | awk '{print $NF}'"
+        fuel_net_cfg_file = admin_remote.check_call(cmd)['stdout'][0].strip()
+        with admin_remote.open(fuel_net_cfg_file, 'r') as f:
+            fuel_net_cfg = yaml.load(f)
+        return fuel_net_cfg['management_vip']
+
+    def _s3cmd_enable_keystone_auth(env):
+        """Enable the Keystone auth backend in RadosGW on all controllers"""
+        logger.debug('Enable the Keystone auth backend in RadosGW on '
+                     'all controllers ')
+
+        def parser_set_value():
+            """set new value in config file"""
+            with node.open(ceph_config_file, 'r') as f:
+                parser = configparser.RawConfigParser()
+                parser.readfp(f)
+                # if value already set -> don't modify and restart radosgw.
+                try:
+                    try_get = parser.get('client.radosgw.gateway',
```

Comment:
The thing here that if value already set, there is no need to to restart `radosgw` (as it may take up to 2 minutes).
But it value not set -> need to restart `radosgw`